crates/typst/src/visualize/gradient.rs

```diff
@@ -1249,7 +1249,11 @@ fn sample_stops(stops: &[(Color, Ratio)], mixing_space: ColorSpace, t: f64) -> C
 
     let (col_0, pos_0) = stops[low - 1];
     let (col_1, pos_1) = stops[low];
-    let t = (t - pos_0.get()) / (pos_1.get() - pos_0.get());
+    let mut delta = pos_1.get() - pos_0.get();
+    if delta == 0.0 {
```

Comment:
can we get into trouble if `delta` isn't equal to `0.0`, but very close to it?

Reply:
The current way this is dealt with seems somewhat arbitrary to me. Wouldn't it make more sense to simply make t either 0.0, 0.5 or 1.0 in the event that two stops are the same?

Maybe I misunderstand what's going on though.

Reply:
I agree. As-is it behaves as if the two stops were actually very far apart (the full `100% = 1.0`).